.github/workflows/build-pack-release.yml

```
@@ -0,0 +1,96 @@
+          with:
+            role-to-assume: ${{ secrets.AWS_ASSUME_ROLE_ARN_RELEASE }}
+            aws-region: us-west-2
+
+        - name: Get Strong-Naming Key
+          run: aws s3api get-object --bucket ${{ secrets.SNK_BUCKET }} --key awsxrayrecorder.dll.snk awsxrayrecorder.dll.snk
+          
+        - name: Install dependencies
+          run: dotnet restore .\sdk\AWSXRayRecorder.sln
+
+        - name: Build solution
+          run: dotnet build .\sdk\AWSXRayRecorder.sln /p:Configuration=Release --no-restore /p:AssemblyOriginatorKeyFile=${{ github.workspace }}\awsxrayrecorder.dll.snk
+
+        - name: Assume signer role
+          uses: aws-actions/configure-aws-credentials@v1
+          with:
+            role-to-assume: ${{ secrets.AWS_ARTIFACT_ACCESS_ROLE_ARN }}
+            aws-region: us-west-2
+
+        - name: Invoke Signing script
+          run: |
+            .\buildtools\sign_files.ps1 -Path .\sdk\src -Filter AWSXRayRecorder.*.dll -Recurse
+          env:
+            UNSIGNED_BUCKET: ${{ secrets.AWS_UNSIGNED_BUCKET_NAME }}
+            SIGNED_BUCKET: ${{ secrets.AWS_SIGNED_BUCKET_NAME }}
+
+        - name: Pack nugets
+          run: |
+            dotnet pack .\sdk\AWSXRayRecorder.sln  --no-build -c Release -o .\Deployment\nuget-packages
+            nuget pack .\sdk\AWSXRayRecorder.nuspec -OutputDirectory .\Deployment\nuget-packages -Exclude **
+
+        - name: Upload nugets as actions artifact
+          uses: actions/upload-artifact@v2
+          with:
+            name: nuget-packages
+            path: Deployment/nuget-packages/
+
+        - name: Assume nuget role
+          uses: aws-actions/configure-aws-credentials@v1
+          with:
+            role-to-assume: ${{ secrets.NUGET_ACCESS_ROLE_ARN }}
+            aws-region: us-west-2
+
+        # TODO: uncomment this step once the rest of the workflow has been verified in prod.
+        #- name: Push packages to Nuget.org
+        #  run: |
+        #    $nugetKey = aws secretsmanager get-secret-value --secret-id ${{ secrets.NUGET_SECRETS_ID }} --region us-west-2 --output text --query SecretString
+        #    nuget push .\Deployment\nuget-packages\*.nupkg -Source https://api.nuget.org/v3/index.json -ApiKey $nugetKey
+            
+        # TODO: Chnage this step to directly create a public release once the workflow has been verified.
+        - name: Create draft release
+          id: create_release
+          uses: actions/create-release@v1
+          env:
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          with:
+            tag_name: 'V${{ github.event.inputs.version }}'
+            release_name: 'Release ${{ github.event.inputs.version }}'
+            body: 'Please refer [change-log](https://github.com/aws/aws-xray-sdk-dotnet/blob/master/CHANGELOG.md) for more details'
+            draft: true
+            prerelease: false
+
+        - name: Clean up
+          run: |
+            rm .\awsxrayrecorder.dll.snk
```

Can we clean this up earlier or request the key later to reduce the time we have it on the machine?

nit:
```suggestion
        - name: Clean up
          run: >
            rm .\awsxrayrecorder.dll.snk
```